File: sprint-backend/src/main/java/com/willmear/sprint/jobs/domain/Job.java
package com.willmear.sprint.jobs.domain;

import com.fasterxml.jackson.databind.JsonNode;
import java.time.Instant;
import java.util.UUID;

public record Job(
        UUID id,
        UUID workspaceId,
        JobType jobType,
        JobStatus status,
        String queueName,
        JsonNode payload,
        Integer attemptCount,
        Integer maxAttempts,
        Instant availableAt,
        Instant lockedAt,
        String lockedBy,
        Instant startedAt,
        Instant completedAt,
        Instant failedAt,
        String errorMessage,
        String errorCode,
        Instant createdAt,
        Instant updatedAt
) {
}

File: sprint-ui/src/types/job.ts
export interface JobSummary {
  id: string;
  workspaceId: string;
  jobType: string;
  status: string;
  attemptCount: number;
  maxAttempts: number;
  availableAt?: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface Job {
  id: string;
  workspaceId: string;
  jobType: string;
  status: string;
  queueName: string;
  payload: unknown;
  attemptCount: number;
  maxAttempts: number;
  availableAt?: string | null;
  lockedAt?: string | null;
  lockedBy?: string | null;
  startedAt?: string | null;
  completedAt?: string | null;
  failedAt?: string | null;
  errorMessage?: string | null;
  errorCode?: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateJobResponse {
  id: string;
  status: string;
  availableAt: string;
  createdAt: string;
}
